10-zwave.js

```diff
@@ -410,7 +410,16 @@ module.exports = function(RED) {
 					) {
 						if (debug) console.log('attempting direct call to OpenZWave API: %s(%s)', msg.topic, payload);
 						try {
-							ozwDriver[msg.topic].apply(ozwDriver, payload);
+							var retval = ozwDriver[msg.topic].apply(ozwDriver, payload);
+							var newmsg = {'topic': msg.topic};
```

Comment:
Currently, it's just completely naive about async calls, and the behavior is as you're describing (aka, kind of disjointed and unintuitive), the results of async calls will pile in a zwave-in node, and the user doesn't have a way to know what was in response to what call.  It can be made to work with all your data going to the right place, but it's not super pretty; it's most useful for synchronous calls at the moment.

I haven't fully wrapped my head around how I'd build a "wait for the callback, then emit returns from that event" mechanism, but at the very least we'd need to refactor event handling to "catch" the callbacks selectively that originated from async calls, have the code somehow know whether to expect a callback for a given call and what that callback would look like, and have a tracking mechanism for every pending callback stored in context (which may take milliseconds or never return at all, if a node never wakes up).

In other words, that seemed like a little too much for me to bite off all at once, but it'll be awesome if we can get there.